src/matrixroomhandler.ts

```diff
@@ -87,7 +87,11 @@ export class MatrixRoomHandler {
             return this.discord.UserSyncroniser.OnUpdateUser(member.user);
         }).then(() => {
             log.info("OnAliasQueried", `Joining ${member.id} to ${roomId}`);
-            return this.joinRoom(this.discord.GetIntentFromDiscordMember(member), roomId);
+            return this.joinRoom(this.discord.GetIntentFromDiscordMember(member), roomId)
+                .then(() => {
+                  // set the correct discord guild name
+                  this.discord.UserSyncroniser.EnsureJoin(member, roomId);
```

Comment:
This looks a bit bizarre. We are joining and then calling ``EnsureJoin``?

Reply:
That is correct, as the normal join doesn't set the display name of the guild correctly. EnsureJoin will also ensure that the member has the correct nickname override for that guild set and, wtih #175 also the role stuff

Reply:
Fair nuff, we probably should rethink that name but let's do that later.